var/spack/repos/builtin/packages/py-fava/package.py

```diff
@@ -0,0 +1,44 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+
+
+class PyFava(PythonPackage):
+    """Fava is a web interface for the double-entry bookkeeping software
+       Beancount with a focus on features and usability."""
+
+    homepage = "https://beancount.github.io/fava/"
+    pypi     = "fava/fava-1.18.tar.gz"
+
+    version('1.18',  sha256='21336b695708497e6f00cab77135b174c51feb2713b657e0e208282960885bf5')
+
+    # For some reason Fava adds a whole bunch of executables to
+    # its bin directory, and this causes clashes when loading
+    # the module.
+    extends('python', ignore='bin/(bean-*|chardetect|cheroot|flask|py*|treeify|upload-to-sheets)')
```

Comment:
Do `chardetect`, `cheroot`, `flask`, and `py*` cause problems with activation? If you just want to ignore everything in bin you can do `ignore='bin/.*'`.

Reply:
Yes they do. There's one binary, `fava`, that I want to not be ignored -- all of the rest should be. I'll try to implement this more elegantly.

Reply:
I've managed to simplify this such that only the `fava` script is considered during package activation. It seems to be working as expected.
```
$ ls <PATH_TO_SPACK>/.../python-3.8.7-6jx33sfut72pvznq3dlpqipvvrkkcqty/bin
2to3             bean-check       bean-extract     bean-identify    bean-report      bottle.py        easy_install     idle3            py.test          pydoc3.8         pyrsa-keygen     pyrsa-verify     python-config    python3.8        treeify
2to3-3.8         bean-doctor      bean-file        bean-price       bean-sql         chardetect       easy_install-3.8 idle3.8          pybabel          pyrsa-decrypt    pyrsa-priv2pub   pytest           python3          python3.8-config upload-to-sheets
bean-bake        bean-example     bean-format      bean-query       bean-web         cheroot          flask            markdown2        pydoc3           pyrsa-encrypt    pyrsa-sign       python           python3-config   python3.8-gdb.py

$ spack activate /vjxnkx5
==> Activating extension py-fava@1.18%apple-clang@12.0.0 arch=darwin-catalina-ivybridge/vjxnkx5 for python@3.8.7%apple-clang@12.0.0+bz2+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tix~tkinter~ucs4+uuid+zlib patches=0d98e93189bc278fbc37a50ed7f183bd8aaf249a8e1670a465f0db6bb4f8cf87 arch=darwin-catalina-ivybridge/6jx33sf

$ ls <PATH_TO_SPACK>/.../python-3.8.7-6jx33sfut72pvznq3dlpqipvvrkkcqty/bin
2to3             bean-check       bean-extract     bean-identify    bean-report      bottle.py        easy_install     flask            markdown2        pydoc3           pyrsa-encrypt    pyrsa-sign       python           python3-config   python3.8-gdb.py
2to3-3.8         bean-doctor      bean-file        bean-price       bean-sql         chardetect       easy_install-3.8 idle3            py.test          pydoc3.8         pyrsa-keygen     pyrsa-verify     python-config    python3.8        treeify
bean-bake        bean-example     bean-format      bean-query       bean-web         cheroot          fava             idle3.8          pybabel          pyrsa-decrypt    pyrsa-priv2pub   pytest           python3          python3.8-config upload-to-sheets
                                                                                                      ^^^^
                                                                                                      ||||
```